src/charm.py

```diff
@@ -59,6 +59,15 @@ def _on_install(self, _: ops.InstallEvent) -> None:
 
     def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
         """Handle config changed event. Update the agent's label in the relation's databag."""
+        logger.info("CONFIG CHANGED")
+        logger.info("%s", self.state.apt_packages)
+        try:
+            self.jenkins_agent_service.install_apt_packages(self.state.apt_packages)
+        except service.PackageInstallError as exc:
+            logger.error("Error installing apt packages %s", exc)
+            self.model.unit.status = ops.BlockedStatus("Error installing apt packages: %s")
```

Comment:
Does `service.PackageInstallError` excludes network-related errors?
I would think network-related error would not need to go to block state.

Reply:
The `apt` lib provided uses subprocess.run under the hood so there's not much ways to differentiate the network related errors. What the lib does expose are errors like `PackageNotFound`, which was suitable for user to unblock by re-configuring the config option.
Do you think ErrorState might be more well-fitting for both cases?

Reply:
For error that requires config change, it should be in block state.
If the `PackageNotFound` only is thrown when user need to unblock it, then it should use `BlockedState`.

For network-related it should be error state.

Reply:
Since network related errors would not be caught (subprocess error would be raised here) - it should fallthrough and go into `ErroredState` automatically.